frictionless/portals/github/manager.py

```diff
@@ -1,45 +1,190 @@
+        base_path = (
+            f"https://raw.githubusercontent.com/{location}/{repository.default_branch}"
+        )
         contents = repository.get_contents("")
-        paths = []
-        while contents:
-            file_content = contents.pop(0)  # type: ignore
-            if file_content.type == "dir":
-                contents.extend(repository.get_contents(file_content.path))  # type: ignore
-            else:
-                paths.append(file_content.path)
-        package = Package()
-        for path in paths:
-            fullpath = f"https://raw.githubusercontent.com/{location}/{branch}/{path}"
-            if path in ["datapackage.json", "datapackage.yaml"]:
-                return Package.from_descriptor(fullpath)
-            if path.endswith(".csv"):
-                package.add_resource(Resource(path=fullpath))
-        return package
+        resource_path = get_resources(contents, repository)
+        package = get_package(resource_path, repository, base_path, self.control.formats)
+        if package.resources:
+            return package
+        note = "Package/s not found"
+        raise FrictionlessException(note)
 
     # Write
-    # TODO: implement
+
+    def write_package(self, package: Package) -> NamedTuple:
+        assert self.control.repo
+        assert self.control.apikey
+
+        # Create repo
+        repository = None
+        user = None
+        try:
+            client = platform.github.Github(self.control.apikey)
+            user = client.get_user()
+            repository = user.create_repo(
+                name=self.control.repo, auto_init=True, gitignore_template="Python"
+            )
+        except Exception as exception:
+            note = "Github API error:" + repr(exception)
+            raise FrictionlessException(note)
+
+        # Write package file to the repo
+        file_created = None
+        content = package.to_json()
+        if content:
+            newfile = self.control.filename or "datapackage.json"
+            repository = user.get_repo(self.control.repo)
+            email = user.email or self.control.email
+            username = self.control.name or user.name or self.control.user
+            assert email
+            assert username
+            try:
+                author = platform.github.InputGitAuthor(username, email)
+                file_created = repository.create_file(
+                    path=newfile,
+                    message="Create package.json",
+                    content=content,
+                    branch=repository.default_branch,
+                    committer=author,
+                    author=author,
+                )
+            except Exception as exception:
+                note = "Github API error:" + repr(exception)
+                raise FrictionlessException(note)
+
+        response = namedtuple("response", ["repository", "file_created"])
+        return response(repository, file_created)
```

Comment:
I think here and in `Packge.to_github` we need to return a full URL to the published data package like: https://raw.githubusercontent.com/fdtester/test-repo-with-datapackage-yaml/master/datapackage.yaml

Reply:
But I'm not yet sure. WDYT?

Reply:
I think returning the package file makes sense. Or may be returning the Repository only and we can use this method of `Repository` to test the package file. 
`repository.get_contents(package_filename)`